app/models/user.rb

```diff
@@ -1,20 +1,15 @@
 class User < ApplicationRecord
-  has_many :posts, foreign_key: 'author_id'
-  has_many :comments, foreign_key: 'author_id'
-  has_many :likes, foreign_key: 'author_id'
-
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :validatable, :confirmable
+  has_many :comments, dependent: :destroy, foreign_key: 'author_id'
+  has_many :likes, dependent: :destroy, foreign_key: 'author_id'
+  has_many :posts, dependent: :destroy, foreign_key: 'author_id'
```

Comment:
- [ ] When I started your application using the `rails s` command, I got the following error:

![image](https://github.com/uchexm/bloga/assets/77106962/d46862cc-e934-4e82-a7bb-7ad5b1827fe9)

I noticed that the `devise` gem was not set up properly in this application. I would advise you to read this [article](https://levelup.gitconnected.com/how-to-add-authentication-to-rails-web-app-using-devise-a5caedbbc093) or this [article](https://makingsenseofrails.dev/how-to-set-up-devise-in-rails-7-72626ead7e0b) to set up devise properly in this project. 👍 💻